specifyweb/specify/tree_views.py

```diff
@@ -342,4 +343,361 @@ def perm_target(tree):
         'storage': StorageMutationPT,
         'geologictimeperiod': GeologictimeperiodMutationPT,
         'lithostrat': LithostratMutationPT,
-    }[tree]
\ No newline at end of file
+    }[tree]
+
+TAXON_RANKS = {
+    'TAXONOMY_ROOT': 0,
+    'KINGDOM': 10,
+    'SUBKINGDOM': 20,
+    'DIVISION': 30,
+    'PHYLUM': 30,
+    'SUBDIVISION': 40,
+    'SUBPHYLUM': 40,
+    'SUPERCLASS': 50,
+    'CLASS': 60,
+    'SUBCLASS': 70,
+    'INFRACLASS': 80,
+    'SUPERORDER': 90,
+    'ORDER': 100,
+    'SUBORDER': 110,
+    'INFRAORDER': 120,
+    'PARVORDER': 125,
+    'SUPERFAMILY': 130,
+    'FAMILY': 140,
+    'SUBFAMILY': 150,
+    'TRIBE': 160,
+    'SUBTRIBE': 170,
+    'GENUS': 180,
+    'SUBGENUS': 190,
+    'SECTION': 200,
+    'SUBSECTION': 210,
+    'SPECIES': 220,
+    'SUBSPECIES': 230,
+    'VARIETY': 240,
+    'SUBVARIETY': 250,
+    'FORMA': 260,
+    'SUBFORMA': 270
+}
+GEOGRAPHY_RANKS = {
+    'CONTINENT': 100,
+    'COUNTRY': 200,
+    'STATE': 300,
+    'COUNTY': 400
+}
+STORAGE_RANKS = {
+    'BUILDING': 100,
+    'COLLECTION': 150,
+    'ROOM': 200,
+    'AISLE': 250,
+    'CABINET': 300,
+    'SHELF': 350,
+    'BOX': 400,
+    'RACK': 450,
+    'VIAL': 500
+}
+GEOLOGIC_TIME_PERIOD_RANKS = {
+    'ERA': 100,
+    'PERIOD': 200,
+    'EPOCH': 300,
+    'AGE': 400
+}
+LITHO_STRAT_RANKS = {
+    'SUPERGROUP': 100,
+    'GROUP': 200,
+    'FORMATION': 300,
+    'MEMBER': 400,
+    'BED': 500
+}
+
+DEFAULT_RANK_INCREMENT = 100
+TAXON_RANK_INCREMENT = 10
+GEOGRAPHY_RANK_INCREMENT = DEFAULT_RANK_INCREMENT
+STORAGE_RANK_INCREMENT = 50
+GEOLOGIC_TIME_PERIOD_INCREMENT = DEFAULT_RANK_INCREMENT
+LITHO_STRAT_INCREMENT = DEFAULT_RANK_INCREMENT
+
+TREE_RANKS_MAPPING = {
+    'taxon': (TAXON_RANKS, TAXON_RANK_INCREMENT),
+    'geography': (GEOGRAPHY_RANKS, GEOGRAPHY_RANK_INCREMENT),
+    'storage': (STORAGE_RANKS, STORAGE_RANK_INCREMENT),
+    'geologictimeperiod': (GEOLOGIC_TIME_PERIOD_RANKS, GEOLOGIC_TIME_PERIOD_INCREMENT),
+    'lithostrat': (LITHO_STRAT_RANKS, LITHO_STRAT_INCREMENT),
+}
+
+@openapi.schema({
+    "post": {
+        "parameters": [
+            {
+                "name": "newRankName",
+                "in": "formData",
+                "required": True,
+                "schema": {
+                    "type": "string"
+                },
+                "description": "The name of the new rank to add"
+            },
+            {
+                "name": "targetRankName",
+                "in": "formData",
+                "required": True,
+                "schema": {
+                    "type": "string"
+                },
+                "description": "The name of the parent rank to add the new rank to (use 'root' to add to the front)"
+            },
+            {
+                "name": "treeType",
```

Comment:
could you call this treeName instead (and for the other endpoint too).
that is the name front-end uses (rather than treeType)
(unless treeType is more common on the back-end)